packages/devextreme/js/common/ai-integration.d.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ * @docid
+ * @namespace DevExpress.aiIntegration
+ * @public
+ */
+export interface Prompt {
+  /**
+   * @docid
+   * @public
+   */
+  system?: string;
+  /**
+   * @docid
+   * @public
+   */
+  user?: string;
+}
+
+/**
+ * @docid
+ * @namespace DevExpress.aiIntegration
+ * @public
+ */
+export interface RequestParams {
+  /**
+   * @docid
+   * @public
+   */
+  prompt: Prompt;
+  /**
+   * @docid
+   * @public
+   */
+  onChunk: (chunk: string) => void;
+}
+
+/**
+ * @docid
+ * @namespace DevExpress.aiIntegration
+ * @public
+ */
+export interface ResponseParams {
+  /**
+   * @docid
+   * @public
+   */
+  promise: Promise<string>;
+  /**
+   * @docid
+   * @public
+   */
+  abort: () => void;
+}
+
+/**
+ * @docid
+ * @namespace DevExpress.aiIntegration
+ * @public
+ */
+export interface TranslateCommandParams {
+  /**
+   * @docid
+   * @public
+   */
+  text: string;
+  /**
+   * @docid
+   * @public
+   */
+  lang: string;
+}
+
+/**
+ * @docid
+ * @namespace DevExpress.aiIntegration
+ * @public
+ */
+export type TranslateCommandResult = string;
+
+/**
+ * @docid
+ * @namespace DevExpress.aiIntegration
+ * @public
+ */
+export type BaseCommandResult = TranslateCommandResult;
```

Comment:
As both aliases point to `string` - maybe we could keep them private for now?
Are their specific cases where they are expected to be used in the end project?

```suggestion
/**
 * @namespace DevExpress.aiIntegration
 */
export type TranslateCommandResult = string;

/**
 * @namespace DevExpress.aiIntegration
 */
export type BaseCommandResult = TranslateCommandResult;
```

Reply:
I added this type here purely because it is used to describe RequestCallbacks, and RequestCallbacks are used to describe AIIntegration.translate. In the final application, the user should only use AIIntegration, AIProvider (described with RequestParams (described with Prompt), ResponseParams. Please, if you think some types should be private - let's make them private.